### PR TITLE
Add "Developing Luanti for Android"

### LINKED ISSUE #110: Add docs for building for Android (e.g. using Android Studio)
The process is quite straightforward, but a step-by-step guide would be nice nevertheless.

I see that @MisterE123 has already published such a guide [on the forums](https://forum.luanti.org/viewtopic.php?t=28869) (though the license is unclear, and it can be simplified now that irrlichtmt is no longer separate). I imagine @rollerozxa might also have some helpful input.

### PR DESCRIPTION
**Pull request by appgurueu**
Closes #110. Feel free to suggest further improvements or additions. I have tested this locally.

### CODE REVIEW

File: content/developing-luanti-for-android.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+## Prerequisites
+
+Basic familiarity with the command line is assumed.
+You have a working network connection.
+You have already downloaded / cloned a copy of Luanti
+and are currently inside the `luanti` project folder.
+You have an Android device you want to install Luanti on for testing purposes
+or you want to produce a Luanti release build for distribution.
+
+## Using CLI tools
```
Comment (rollerozxa) [inline content/developing-luanti-for-android.md:21]: You would want to mention how to actually set up the Android command-line tools, agreeing to the license terms and setting the local.properties path. (See [these instructions](https://principia-web.se/wiki/Compiling_Principia#android) for an example)

File: content/developing-luanti-for-android.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+./app/build/outputs/apk/debug/app-armeabi-v7a-debug.apk
+./app/build/outputs/apk/debug/app-x86_64-debug.apk
+./app/build/outputs/apk/debug/app-x86-debug.apk
+./app/build/outputs/apk/debug/app-arm64-v8a-debug.apk
+```
+
+Now that you have an APK, you can use the [Android Debug Bridge (ADB)](https://developer.android.com/tools/adb) to install it [^1] via
+
+```sh
+adb install ./app/build/outputs/apk/debug/app-arm64-v8a-debug.apk
```
Comment (rollerozxa) [inline content/developing-luanti-for-android.md:49]: `./gradlew installDebug` should do the same and will detect the ABI of the device it is connected to via ADB.

File: content/developing-luanti-for-android.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+# Developing Luanti for Android
+
+The instructions in this article were written for and tested on Linux,
+but the tools used here should also work on other platforms.
+
+## Prerequisites
+
+Basic familiarity with the command line is assumed.
+You have a working network connection.
```
Comment (rollerozxa) [inline content/developing-luanti-for-android.md:15]: Should this be a list? When rendered to markdown all the lines get squished into one paragraph.

Comment (appgurueu) [reply content/developing-luanti-for-android.md:15]: I think either is fine but sure, a list may be nicer.

File: content/developing-luanti-for-android.md (hunk 1 above)
Comment (appgurueu) [reply content/developing-luanti-for-android.md:21]: Right, you linked that to me before. Thanks for reminding me.

Comment (appgurueu) [reply content/developing-luanti-for-android.md]: Alright, I've added an ~~botched~~ abridged version of your instructions. Do you think it's fine in its current state?

Comment (rollerozxa) [reply content/developing-luanti-for-android.md]: Yes, looks okay.

**Pull request merged**